modules/apps/translation/translation-service/src/main/java/com/liferay/translation/internal/translator/TranslatorRegistryImpl.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.translation.internal.translator;
+
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerList;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerListFactory;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.translation.translator.Translator;
+import com.liferay.translation.translator.TranslatorRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Alicia Garcia
+ */
+@Component(service = TranslatorRegistry.class)
+public class TranslatorRegistryImpl implements TranslatorRegistry {
+
+	@Override
+	public Translator getCompanyTranslator(long companyId)
+		throws ConfigurationException {
+
+		List<Translator> translators = _getTranslators(companyId);
+
+		if (translators.isEmpty()) {
+			return null;
+		}
+
+		return translators.get(0);
+	}
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerList = ServiceTrackerListFactory.open(
+			bundleContext, Translator.class);
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_serviceTrackerList.close();
+	}
+
+	private List<Translator> _getTranslators(long companyId)
+		throws ConfigurationException {
+
+		List<Translator> translators = new ArrayList<>();
+
+		for (Translator translator : _serviceTrackerList) {
+			if (translator.isEnabled(companyId)) {
+				translators.add(translator);
+			}
+		}
+
+		return translators;
+	}
```

Comment:
Can you inline this method? Right now, we're constructing a list with all candidates, but we could stop looking as soon as we found one:
```
public Translator getCompanyTranslator(long companyId) {
		for (Translator translator : _serviceTrackerList) {
			if (translator.isEnabled(companyId)) {
				return translator;
			}
		}

		return null;
}
```
Sorry I didn't mention this before.

Reply:
This is related with my previous comment : https://github.com/liferay-lima/liferay-portal/pull/2438#discussion_r741263487 I was thinking on return a method that can return all the translators.

but as previously said. It can be done in the future if needed.

thanks!!